RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc

```diff
@@ -933,41 +1010,36 @@ bool SiPixelTemplate2D::xytemp(float xhit, float yhit, bool ydouble[BYM2], bool
    
    //  First zero the local 2-d template
    
-   for(j=0; j<BXM2; ++j) {for(i=0; i<BYM2; ++i) {xytemp_[j][i] = 0.f;}}
+   for(int j=0; j<BXM2; ++j) {for(int i=0; i<BYM2; ++i) {xytemp_[j][i] = 0.f;}}
    
    // Loop over the non-zero part of the template index space and interpolate
    
-   for(j=jmin_; j<=jmax_; ++j) {
+   for(int j=jmin_; j<=jmax_; ++j) {
       // Flip indices as needed
       if(flip_x_) {jflipx=T2XSIZE-1-j; m = deltax+jflipx;} else {m = deltax+j;}
-      for(i=imin_; i<=imax_; ++i) {
+      for(int i=imin_; i<=imax_; ++i) {
          if(flip_y_) {iflipy=T2YSIZE-1-i; n = deltay+iflipy;} else {n = deltay+i;}
          if(m>=0 && m<=BXM3 && n>=0 && n<=BYM3) {
-            tmpxy = entry00_->xytemp[k0][l0][i][j]
-            + adx*(entry00_->xytemp[k0][l1][i][j] - entry00_->xytemp[k0][l0][i][j])
-            + ady*(entry00_->xytemp[k1][l0][i][j] - entry00_->xytemp[k0][l0][i][j])
-            + adcota_*(entry01_->xytemp[k0][l0][i][j] - entry00_->xytemp[k0][l0][i][j])
-            + adcotb_*(entry10_->xytemp[k0][l0][i][j] - entry00_->xytemp[k0][l0][i][j]);
-            if(tmpxy > 0.f) {
-               xytemp_[m][n] = tmpxy;
-            } else {
-               xytemp_[m][n] = 0.f;
-            }
+            xytemp_[m][n] = (float)entry00_->xytemp[k0][l0][i][j]

```

Comment:
This I need to check with Morris.  He's left for the day, so I'll let you know tomorrow.

Reply:
From Morris:

The lines in question have to do with the interpolation calculation for each pixel in the 2D array.  Physically, trapped holes can (and do) induce negative signals in pixels near the edges of the cluster.  Since we readout only positive signals, these are not important for most things.  However, now that we are iteratively fitting 2D clusters to reconstruct them, the pixel cell derivatives are needed.  Shifting a pixel near the boundary involves a slope and the presence of negative signals increases the slope, so storing negative numbers should improve accuracy.  Additionally, putting un-needed if statements in loops probably ruins vectorization.